### PR TITLE
EKA-G-CM-33 API consent-requests/init

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5622591&bid=18417928)

File: src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java [hunk 1]
```diff
@@ -66,408 +71,458 @@
...
+
+    private Mono<Void> validateHiTypes(HIType[] hiTypes) {
+        return conceptValidator.validateHITypes(Arrays.stream(hiTypes).map(type -> type.getValue()).collect(Collectors.toList()))
+                .flatMap(result ->
+                        result.booleanValue() ? Mono.empty()
+                                : Mono.error(new ClientError(BAD_REQUEST,
+                                new ErrorRepresentation(new Error(INVALID_HITYPE, "Invalid HI Type"))))
+                );
+    }
+
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java:160]: Whatever refactoring I had done is all gone when you rebased with your branch. Can you fix that? Check master?

Not just this method, there are other methods too.

File: src/main/java/in/projecteka/consentmanager/consent/ConsentRequestController.java [hunk 2]
```diff
@@ -83,11 +91,11 @@ private static RequestCreatedRepresentation buildResponse(String requestId) {
...
-						consentManager
-								.approveConsent(caller.getUsername(), requestId, consentApprovalRequest.getConsents())
-								.doOnSuccess(discard -> {
-									logger.debug("[approve] putting {} in used tokens", caller.getSessionId());
-									usedTokens.put(caller.getSessionId(), "").subscribe();
+						consentManager.
+								approveConsent(caller.getUsername(), requestId, consentApprovalRequest.getConsents())
+								.flatMap(response -> {
+									logger.debug("[approve] putting {} in used tokens",caller.getSessionId());
+									return usedTokens.put(caller.getSessionId(),"").thenReturn(response);
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentRequestController.java:98]: See the same thing again?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5622591&bid=18417928)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5622591&bid=18417928)

File: src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java (hunk 1 above)
Comment (Saisree9) [reply src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java]: Fixed. Something went wrong on rebasing. All the methods are up to date with master now.
Thank you

File: src/main/java/in/projecteka/consentmanager/consent/ConsentRequestController.java [hunk 3]
```diff
@@ -47,6 +47,14 @@ protected void initBinder(WebDataBinder binder) {
...
 	}
 
+    @PostMapping(value = "/v1/consent-requests/init")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public Mono<Void> initConsentRequest(
+            @RequestBody @Valid @ModelAttribute("consentRequest") ConsentRequest request) {
+        return consentManager.requestConsent(request.getConsent(), request.getRequestId());
+    }
+
+
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentRequestController.java:57]:
```suggestion
```

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5622591&bid=18417928)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5622591&bid=18417928)

**Pull request merged**